<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Genomics Visualization: Background & Prior Work • Aprameya Mysore</title>
    <meta name="description" content="There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization.

">
    <meta name="keywords" content="thesis">
    
    	<!-- Twitter Cards -->
	<meta name="twitter:title" content="Genomics Visualization: Background & Prior Work">
	<meta name="twitter:description" content="There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization.

">
	<meta name="twitter:site" content="@invizibility">
	<meta name="twitter:creator" content="@invizibility">
	
	<meta name="twitter:card" content="summary">
	<meta name="twitter:image" content="https://invizibility.github.io/images/">
	
	<!-- Open Graph -->
	<meta property="og:locale" content="en">
	<meta property="og:type" content="article">
	<meta property="og:title" content="Genomics Visualization: Background & Prior Work">
	<meta property="og:description" content="There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization.

">
	<meta property="og:url" content="https://invizibility.github.io/blog/background/">
	<meta property="og:site_name" content="Aprameya Mysore">

    <link rel="canonical" href="https://invizibility.github.io/blog/background/">
    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

    <link href="https://invizibility.github.io/atom.xml" type="application/atom+xml" rel="alternate" title="Aprameya Mysore Atom Feed">
    <link href="https://invizibility.github.io/sitemap.xml" type="application/xml" rel="sitemap" title="Sitemap">

    <meta name="HandheldFriendly" content="True">
    <meta name="MobileOptimized" content="320">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="cleartype" content="on">

    <link rel="stylesheet" href="https://invizibility.github.io/css/main.css">
    <!-- HTML5 Shiv and Media Query Support for IE -->
    <!--[if lt IE 9]>
      <script src="https://invizibility.github.io/js/vendor/html5shiv.min.js"></script>
      <script src="https://invizibility.github.io/js/vendor/respond.min.js"></script>
    <![endif]-->

  </head>

  <body id="js-body">
    <!--[if lt IE 9]><div class="upgrade notice-warning"><strong>Your browser is quite old!</strong> Why not <a href="http://whatbrowser.org/">upgrade to a newer one</a> to better enjoy this site?</div><![endif]-->

    <header id="masthead">
  <div class="inner-wrap">
    <a href="https://invizibility.github.io/" class="site-title">Aprameya Mysore</a>
    <nav role="navigation" class="menu top-menu">
        <ul class="menu-item">
	<li class="home"><a href="/">Aprameya Mysore</a></li>
	

		
		
			<li><a href="https://invizibility.github.io" ></a></li>
		
		

	

		
		
			<li><a href="https://invizibility.github.io/contact/" >@</a></li>
		
		

	

		
		
			<li><a href="https://invizibility.github.io/blog/" >blog</a></li>
		
		

	

		
		
			<li><a href="https://invizibility.github.io/cv/" >cv</a></li>
		
		

	

		
		
			<li 
				
					><a href="https://invizibility.github.io#content" >work</a>
					                                                                                    				
			</li>
			
		

		
</ul>
    </nav>
  </div><!-- /.inner-wrap -->
</header><!-- /.masthead -->
    <!--<nav role="navigation" id="js-menu" class="sliding-menu-content">
  <h5>Aprameya Mysore <span>Table of Contents</span></h5>
  <ul class="menu-item">
    <li>
      <a href="https://invizibility.github.io">
        
        <div class="title"></div>
        
      </a>
    </li><li>
      <a href="https://invizibility.github.io/contact/">
        
        <div class="title">@</div>
        
      </a>
    </li><li>
      <a href="https://invizibility.github.io/blog/">
        
        <div class="title">blog</div>
        
      </a>
    </li><li>
      <a href="https://invizibility.github.io/cv/">
        
        <div class="title">cv</div>
        
      </a>
    </li><li>
      <a href="https://invizibility.github.io">
        
        <div class="title">work</div>
        
      </a>
    </li>
  </ul>
</nav>
<button type="button" id="js-menu-trigger" class="sliding-menu-button lines-button x2" role="button" aria-label="Toggle Navigation">
  <span class="nav-lines"></span>
</button>

<div id="js-menu-screen" class="menu-screen"></div>
-->

    <div id="page-wrapper">
      <div id="main" role="main">
	<article class="wrap" itemscope itemtype="http://schema.org/Article">
		
		
  <nav class="breadcrumbs">
    <span itemscope itemtype="http://data-vocabulary.org/Breadcrumb">
      <a href="https://invizibility.github.io" itemprop="url">
        <span itemprop="title">Home</span>
      </a> › 
    <span itemscope itemtype="http://data-vocabulary.org/Breadcrumb">
      <a href="https://invizibility.github.io/blog/" itemprop="url">
        <span itemprop="title">Blog</span>
      </a>
    </span>
  </nav><!-- /.breadcrumbs -->

		<div class="page-title">
			<h1>Genomics Visualization: Background & Prior Work</h1>
		</div>
		<div class="inner-wrap">
			<div id="content" class="page-content" itemprop="articleBody">
				<p>There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization.</p>

<ul>
  <li>media consisting of visualizations that describe a problem, process, or observation relating to genomic data.</li>
  <li>media describing the problem space of visualization in genomics. this has to do with genomics’ place within the epistemological regime of systems biology</li>
  <li>media that attempts to create criteria for good/bad visualizations</li>
</ul>

<p>In this post I will tackle the first area. I’ve tried to select visualizations that I think are either effective or ineffective for notable reasons; there are a lot of bad visualizations out there so it is important to learn from ‘near misses’ instead of trying to extract meaning from something that is a complete failure.</p>

<p><a href="https://petridishtalk.com/2012/06/30/gene-visualization/">This</a> blog post by Mo Rahman, the founder of <a href="http://genedrop.com/">GeneDrop</a>, provides a good overview of the most common types of visual grammar used in contemporary genomics problem solving. I want to elaborate on this overview of genomics visualization strategies by organizing visualizations by their functional goals in addition to just their overt appearance.</p>

<h3 id="track-visualization">Track Visualization</h3>

<p>Track visualizations present sequential information in a horizontal extended track, and are primarily utilized to examine issues of alignment between multiple sequences. The genome alignment problem space ranges from analyzing small varations in sequence amongst members of a population or species, to asking questions about complex conserved trends across species or generations. Given the scale and complexity of the variety of alignment relationships potentially of interest, it can be very trickly to legibly visualize an observation or a hypothesis. The following figure shows similarity and inversion of particular regions across 11 <em>Lactobacillus</em> genomes represented as tracks, from the paper <a href="http://journal.frontiersin.org/article/10.3389/fmicb.2012.00048/full">“Tracing lifestyle adaptation in prokaryotic genomes”</a>. It gets quite messy.</p>

<p><img src="/images/blog/2016-09-18-background/alignmentmess.jpg" alt="alignmentmess" /></p>

<p>As such, researchers have explored custom track visualization solutions when a linear genome representation is required. A couple interesting examples include:</p>

<h5 id="genedrop-custom-browser">GeneDrop Custom Browser</h5>

<p>This browser has been shown by Mo Rahman on Twitter and on his blog. It’s a custom in house genome browser used for informatics work at GeneDrop, and effectively shows how a well designed track visualization can frame the complexity at various levels of detail present in a multiple sequence alignment analysis. Here it is shown in a workflow for resolving structural variations in certain repetetive regions of genome sequence</p>

<p><img src="/images/blog/2016-09-18-background/GeneDropProprietaryBrowser.png" alt="GeneDrop" /></p>

<p>and again here in an interesting <a href="https://petridishtalk.com/2014/10/02/retooling-analysis-pipelines-from-human-to-ebov-ngs-data-for-rapid-alignment-and-strain-identification/">post</a> about analyzing strain variations of the ebola virus</p>

<p><img src="/images/blog/2016-09-18-background/GeneDrop2.png" alt="GeneDrop2" /></p>

<h5 id="sequence-surveyor">Sequence Surveyor</h5>

<p>This tool was developed at the Computer Graphics department at the University of Wisconsin-Madison as a scalable solution for comparing and clustering alignments. It uses gradients and cluster averaging to provide a way to compare multiple sequence alignments like in the <em>Lactobacillus</em> example, without the spiderweb of edges required to show congruence/inversion. This tool actually actually has interesting applications in text corpus analysis, but that is a discussion for a separate post.</p>

<p><img src="/images/blog/2016-09-18-background/sequencesurv.gif" alt="SeqSurv" /></p>

<h5 id="bactogenie">BactoGENIE</h5>

<p>This project is interesting because it unifies a hardware visualization solution with a custom visual grammar. BactoGENIE was created for visualizing metagenomic datasets, which are sequencing data taken from a variety of species populating a certain local biome (water in a lake, your intestines, etc.) Figuring out similarities between all the different types of bacterial DNA in the sample yields powerful insights about the stability and the history a local biomic environment. BactoGENIE runs on a large immersive display at the Experimental Visualization Laboratory at the University of Chicago</p>

<p><img src="/images/blog/2016-09-18-background/bactogenie.jpg" alt="Bact" /></p>

<h4 id="ucsc">UCSC</h4>

<p>The <a href="genome.ucsc.edu">UCSC genome browser</a> provides the most widely used and most accessible track visualization platform for genomic data. This project began to showcase the initial findings of the Human Genome Project, and continues to be a widely used repository for genomic alignment and annotation data for several different species. The UCSC browser actually combines data that are traditionally seen in heatmaps (such as measurement of gene expression, protein binding affinity, etc) along the distance of the sequence in addition to the tracks which indicate locations of genes/promoter regions for various species. While seeing these two types of information alongside each other can be very powerful analytically, the sheer quantity of informational tracks available on UCSC makes it difficult to explore potentially interesting patterns across a sequence. As such, the UCSC browser is used primarily to look up sections of sequence that have already been identified as noteworthy.</p>

<p><img src="/images/blog/2016-09-18-background/ucsc.JPG" alt="alt" /></p>

<h3 id="heatmap-visualization">Heatmap Visualization</h3>

<p>Heatmap visualizations are used for a variety of purposes in genomic analysis. When understanding the range of gene products created by a certain genome, researchers will use heatmaps to visualize patterns occuring in various forms of microarray data across specimens. This heatmap from <a href="http://journal.frontiersin.org/article/10.3389/fonc.2012.00065/full">Allard et. al 2012</a> shows clustered patterns of 66 transcripts differentially expressed between two populations in a cancer study.</p>

<p><img src="/images/blog/2016-09-18-background/expressionheatmap.jpg" alt="Express" /></p>

<h4 id="igv">IGV</h4>

<p>The <a href="http://software.broadinstitute.org/software/igv/">Integrative Genomics Viewer</a> is a very widely used interface for aggregating and organizing heatmap data. It was created by the Broad Institute in 2011 to allow analysis of diverse datasets that annotate a genomic sequence across multiple scales.</p>

<p><img src="/images/blog/2016-09-18-background/igv_cherry.gif" alt="igv" /></p>

<p>Heatmaps are also used to show relationships of adjacency and connectedness. Additional explorations into microarray data like the one above look not only at how genes are differentially expressed amongst various specimens in an experiment, but how genes are coexpressed with other genes. In this way the heatmap is a reprojection of a graph whereby the rows and columns are the nodes and the color value in the cells represents strength of links. This graphic from <a href="http://circgenetics.ahajournals.org/content/4/1/26.full">Dewey et. al 2010</a> shows such a reprojection.</p>

<p><img src="/images/blog/2016-09-18-background/coexpression.jpg" alt="coexp" /></p>

<p>Another way heatmaps are used to project data about adjacency is in studies of chromosomal conformation. Recent technological breakthroughs have allowed researchers to capture the 3 dimensional structure of a genome inside a cell with higher degrees of confidence, and the visualization of this data is driving the study of how genomic structure affects its function. This graphic from <a href="http://www.cell.com/fulltext/S0092-8674(14)01497-4">Rao et. al 2014</a> utilizes heatmaps in such a way to explain folding of chromatin loops.</p>

<p><img src="/images/blog/2016-09-18-background/hic.jpg" alt="hic" /></p>

<h4 id="juicebox">JuiceBox</h4>

<p><a href="http://www.aidenlab.org/juicebox/">Juicebox</a> is a software for visualizing datasets that describe structural conformation of genomes. Like IGV, it provides dynamic rescaling of data at different zoom levels, which allows a user to observe interesting features across various orders of magnitude. It is currently one of the premier toolkits for structure-function research in genomics.</p>

<p><img src="/images/blog/2016-09-18-background/straight_edge.png" alt="juicebox" /></p>

<h3 id="tree-visualization">Tree Visualization</h3>

<p>Visualizations describing the structure of biological classifications or ‘trees of life’ are amongst the most nascent forms of scientific visualization. Phylogenetic and taxonomic visualizations showcase evolutionary relationships amongst a population of biological entities. The visualized phylogeny provides useful information for a variety of populations, which could include microbial communities local to a biome, families of proteins present across species, or even more broadly defined groups in which relationships of common ancestry and heredity are of interest. Modern approaches to visualizing phylogenies typically utilize circular dendrograms, which provide increased space for more granular classification at the edges.</p>

<h4 id="graphlan">GraPhlAn</h4>

<p><a href="https://huttenhower.sph.harvard.edu/graphlan">GraPhlAn</a> is a genome phylogeny and taxonomy visualization package based around circular dendrograms. This particular figure from the GraPhlAn paper, <a href="https://peerj.com/articles/1029/">Asnicar et. al 2015</a>, demonstrates how the radial structure of the tree can be used to encode both information about hierarchy in a population of bacterial genomes, as well as functional attributes characterizing different regions of the genomic hierarchy.</p>

<p><img src="/images/blog/2016-09-18-background/phylotree.jpg" alt="phylo" /></p>

<p>Earlier in the paper the authors present this meta-diagram which demonstrates the process by which a dendrogram can be generated from a hierarchical data source. These types of meta-diagrams give an interesting peek into the construction of a design vocabulary for a given visualization problem, and will be covered more extensively in a future blog post.</p>

<p><img src="/images/blog/2016-09-18-background/phylometa.png" alt="phylometa" /></p>

<h4 id="anvio">Anvi’o</h4>

<p><a href="http://merenlab.org/software/anvio/">Anvi’o</a> is a circular dendogram visualization designed as part of a metagenomics analysis pipeline. This pipeline is used to explore how genomic changes across multiple species in an environmental sample are related to broader environmental changes. The visualizations generated by Anvi’o are similar to the dendrograms seen in GraPhlAn, with a notable difference being Anvi’o is designed for encoding substantially more functional information by using the radial axis to represent time.</p>

<p><img src="/images/blog/2016-09-18-background/anvio.png" alt="anvio" /></p>

<h3 id="network-visualization">Network Visualization</h3>

<p>There are a multitude of uses for network visualizations in genomics. Connectivity is a very important theme across genomics, and questions of connectedness are frequent avenues for interrogating a variety of different genomic data. As such, projection and reprojection of various types of genomic data into node-link space is common practice in contemporary genomic visual grammar. Furthermore, networks are intimately related to the epistemological task of genomics within the regime of systems biology: how do characterstics of the genome give rise to function and behavior of other biological systems?</p>

<h4 id="circos">Circos</h4>

<p><a href="http://www.circos.ca/">Circos</a> is a visualization package that is used to make a multitude of chord diagram based plots. Although it is purportedly agnostic to data source, it was designed as a utility for visualizing connectivity and adjacency in genomic data. I see the chord diagram as a reprojected heatmap. The following figure from <a href="http://link.springer.com/article/10.1007/s11032-013-9941-x">Wei et. al</a> shows a circos plot to describe connectivity amongst of group of protein kinases in maize. There are multiple types of functional annotation for each protein kinase visualized radially on the chord diagram as well.</p>

<p><img src="/images/blog/2016-09-18-background/circosmaize.png" alt="circosmaize" /></p>

<h4 id="ayasdi">Ayasdi</h4>

<p>Networks visualizations can also be utilized as a low dimensional embedding of a clustering operation performed in high dimensional space. <a href="https://www.ayasdi.com/blog/office-of-the-cdo/when-is-breast-cancer-like-ovarian-cancer/">Ayasdi</a> utilizes network visualizations to describe the results of topological data analysis performed on high dimensional data of various types. The figure here is comparing clustering of genomic expression profiles of various tumor biopsy samples. The relative shape and color of each network gives a viewer an idea of how similar any two tumors might be for a variety of different cancers. Between the node-link structure and the color gradient, a lot of information can be embedded in this visual style.</p>

<p><img src="/images/blog/2016-09-18-background/breast_cancer_data1.png" alt="ayasdi" /></p>

<h4 id="valence">Valence</h4>

<p>Valence was a design experiment that was part of Ben Fry’s Master’s Thesis: Organic Information Design. Valence sought to explore how networks can function as reprojections of linear sequences, and can potentially elucidate questions about connectedness of phrases that are not apparent in more traditional track based visualizations. It was utilized both on large corpuses of text as well as genomic data.</p>

<p><img src="/images/blog/2016-09-18-background/valence.jpg" alt="valence" /></p>

<h4 id="kegg">KEGG</h4>

<p>From the Kyoto Encyclopedia of Genes and Genomes website, a KEGG pathway is desribed as a manually drawn map representing knowledge of molecular interaction and reaction networks linking data on genomics, transcriptomics, proteomics, and metabolomics. These pathway maps describe higher order systemic functions which implicate genomics into the larger descriptive paradigms of systems biology. While they do not provide the same type of exact mapping from data space to visual space that other modes of genomic visualization do, they are still very important as heuristic tools in systems biology. The following diagram is of KEGG pathway hsa01100, which describes the sum total of metabolic processes that drive human cellular activity.</p>

<p><img src="/images/blog/2016-09-18-background/kegg_global.png" alt="kegg" /></p>

<h3 id="multi-visualization-suites">Multi Visualization Suites</h3>

<p>Sometimes being able to switch between multiple views of a dataset in succession is useful for understanding distributional characteristics and ascertaining significance of outliers.</p>

<h4 id="phinch">Phinch</h4>

<p><a href="http://phinch.org/">Phinch</a> was created by the data visualization studio Pitch Interactive as a tool for visualizing high throughput metagenomic datasets, which contain genomic data from a variety of different organisms present in a local environment. In addition to providing a curated selection of plots to dive into a given dataset, particular attention is paid to providing a ‘designed’ data load in menu. Using data visualization to allow a user to identify which components of a vast data repository are to be visualized in further detail is a interesting and widely useful application of visual grammar in genomics.</p>

<p><img src="/images/blog/2016-09-18-background/viz_gallery_phinch.png" alt="phinch" /></p>

<h4 id="isometric-haplotype-blocks">Isometric Haplotype Blocks</h4>

<p>Ben Fry designed this utility as part of his PhD thesis on computational information design. The study of haplotypes, which are groups of polymorphisms in the genome of a species that are often found together, is critical to understanding what types of genetic variation are related to disease. Understanding haplotype membership can be tricky, and requires identifying correlations amongst groups of polymorphisms in a population that pass a certain threshold known as Linkage Disequilibrium (LD). If there is no significant mechanism causing associations between polymorphisms in one region of a genome and another, population data should reflect that all combinations of polymorphisms in region A and region B are equally probable to occur in a specimen through normal genetic variation. However if region A and B are in strong LD, specific types of polymorphisms in region A are strongly correlated with specific polymorphisms in region B.</p>

<p>Fry’s visualization is an exercise in understanding how data on polymorphism and LD in a given population can help construct an idea of haplotype clustering for that population. By providing multiple isomorphic views of the dataset, the user can maintain context while considering different types of comparative evaluations of the patterns of polymorphism.</p>

<p><img src="/images/blog/2016-09-18-background/isometricblocks-900.jpg" alt="isometric" /></p>

				<hr />
				<footer class="page-footer">
					

<div class="author-image">
	<img src="https://invizibility.github.io/images/" alt="">
</div><!-- ./author-image -->
<div class="author-content">
	<h3 class="author-name" >Written by <span itemprop="author"></span></h3>
	<p class="author-bio"></p>
</div><!-- ./author-content -->
					<div class="inline-btn">
	<a class="btn-social twitter" href="https://twitter.com/intent/tweet?text=Genomics%20Visualization:%20Background%20&amp;%20Prior%20Work&amp;url=https://invizibility.github.io/blog/background/&amp;via=invizibility" target="_blank"><i class="fa fa-twitter" aria-hidden="true"></i> Share on Twitter</a>
	<a class="btn-social facebook" href="https://www.facebook.com/sharer/sharer.php?u=https://invizibility.github.io/blog/background/" target="_blank"><i class="fa fa-facebook" aria-hidden="true"></i> Share on Facebook</a>
	<a class="btn-social google-plus"  href="https://plus.google.com/share?url=https://invizibility.github.io/blog/background/" target="_blank"><i class="fa fa-google-plus" aria-hidden="true"></i> Share on Google+</a>
</div><!-- /.share-this -->

					<div class="page-meta">
	<p>Updated <time datetime="2016-09-18T00:00:00Z" itemprop="datePublished">September 18, 2016</time></p>
</div><!-- /.page-meta -->
				</footer><!-- /.footer -->
				<aside>
					
				</aside>
			</div><!-- /.content -->
		</div><!-- /.inner-wrap -->
		
	</article><!-- ./wrap -->
</div><!-- /#main -->

      <footer role="contentinfo" id="site-footer">
	<nav role="navigation" class="menu bottom-menu">
		<ul class="menu-item">
		
      
			<li><a href="https://invizibility.github.io" ></a></li>
		
		</ul>
	</nav><!-- /.bottom-menu -->
	<p class="copyright">&#169; 2017 <a href="https://invizibility.github.io">Aprameya Mysore</a> powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> + <a href="http://mmistakes.github.io/skinny-bones-jekyll/" rel="nofollow">Skinny Bones</a>.</p>
</footer>
    </div>

    <script src="https://invizibility.github.io/js/vendor/jquery-1.9.1.min.js"></script>
    <script src="https://invizibility.github.io/js/vendor/picturefill.js"></script>
    <script src="https://invizibility.github.io/js/main.js"></script>
    
  </body>

</html>
