<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Background &amp; Prior Work | Aprameya&#39;s Thesis Blog</title>

  
  <meta name="author" content="John Doe">
  

  
  <meta name="description" content="There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization. 

media consisti">
  

  
  
  <meta name="keywords" content="">
  

  <meta id="viewport" name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no, minimal-ui">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">

  <meta property="og:title" content="Background &amp; Prior Work"/>

  <meta property="og:site_name" content="Aprameya&#39;s Thesis Blog"/>

  
  <meta property="og:image" content="/blog/favicon.ico"/>
  

  <link href="/blog/favicon.ico" rel="icon">
  <link rel="alternate" href="/blog/atom.xml" title="Aprameya&#39;s Thesis Blog" type="application/atom+xml">
  <link rel="stylesheet" href="/blog/css/style.css" media="screen" type="text/css">
</head>


<body>
<div class="blog">
  <div class="content">

    <header>
  <div class="site-branding">
    <h1 class="site-title">
      <a href="/blog/">Aprameya&#39;s Thesis Blog</a>
    </h1>
    <p class="site-description"></p>
  </div>
  <nav class="site-navigation">
    <ul>
      
        <li><a href="/">Home</a></li>
      
        <li><a href="/archives">Archives</a></li>
      
    </ul>
  </nav>
</header>

    <main class="site-main posts-loop">
    <article>

  
    
    <h3 class="article-title"><span>Background &amp; Prior Work</span></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2016/09/18/background/" rel="bookmark">
        <time class="entry-date published" datetime="2016-09-18T23:27:58.000Z">
          2016-09-18
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization. </p>
<ul>
<li>media consisting of visualizations that describe a problem, process, or observation relating to genomic data.</li>
<li>media describing the problem space of visualization in genomics. this has to do with genomics’ place within the epistemological regime of systems biology </li>
<li>media that attempts to create criteria for good/bad visualizations</li>
</ul>
<p>In this post I will tackle the first area. I’ve tried to select visualizations that I think are either effective or ineffective for notable reasons; there are a lot of bad visualizations out there so it is important to learn from ‘near misses’ instead of trying to extract meaning from something that is a complete failure. </p>
<p><a href="https://petridishtalk.com/2012/06/30/gene-visualization/" target="_blank" rel="external">This</a> blog post by Mo Rahman, the founder of <a href="http://genedrop.com/" target="_blank" rel="external">GeneDrop</a>, provides a good overview of the most common types of visual grammar used in contemporary genomics problem solving. Going through each of the 5 categories outlined in his post, I’ve aggregated some visualization strategies I think are interesting.</p>
<h3 id="Track-Visualization"><a href="#Track-Visualization" class="headerlink" title="Track Visualization"></a>Track Visualization</h3><p>The <a href="genome.ucsc.edu">UCSC genome browser</a> is considered the standard of this type of visualization. This project began to showcase the initial findings of the Human Genome Project, and continues to be a widely used repository for genomic alignment and annotation data for several different species.</p>
<p><img src="/blog/images/ucsc.JPG" alt="alt"></p>
<p>Track visualizations present sequential information in a horizontal extended track, and invite users to compare alignments between multiple sequences, or annotations (such as measurement of gene expression, or presence of a promoter region) along the distance of the sequence. This is useful for looking up the significance of a particular region on a genome, but for asking questions about complex conserved trends across species or generations, simple linear track visualization becomes complicated. The following figure shows similarity and inversion of particular regions across 11 <em>Lactobacillus</em> genomes represented as tracks, from the paper <a href="http://journal.frontiersin.org/article/10.3389/fmicb.2012.00048/full" target="_blank" rel="external">“Tracing lifestyle adaptation in prokaryotic genomes”</a>. It gets quite messy.</p>
<p><img src="/blog/images/alignmentmess.jpg" alt="alignmentmess"></p>
<p>As such, researchers have explored custom track visualization solutions when a linear genome representation is required. A couple interesting examples include:</p>
<h4 id="GeneDrop-Custom-Browser"><a href="#GeneDrop-Custom-Browser" class="headerlink" title="GeneDrop Custom Browser"></a>GeneDrop Custom Browser</h4><p>This browser has been shown by Mo Rahman on Twitter and on his blog. It’s a custom in house genome browser used for informatics work at GeneDrop, and effectively shows how a well designed track visualization can frame the complexity at various levels of detail present in a multiple sequence alignment analysis. Here it is shown in a workflow for resolving structural variations in certain repetetive regions of genome sequence</p>
<p><img src="/blog/images/GeneDropProprietaryBrowser.png" alt="GeneDrop"></p>
<p>and again here in an interesting <a href="https://petridishtalk.com/2014/10/02/retooling-analysis-pipelines-from-human-to-ebov-ngs-data-for-rapid-alignment-and-strain-identification/" target="_blank" rel="external">post</a> about analyzing strain variations of the ebola virus</p>
<p><img src="/blog/images/GeneDrop2.png" alt="GeneDrop2"></p>
<h4 id="Sequence-Surveyor"><a href="#Sequence-Surveyor" class="headerlink" title="Sequence Surveyor"></a>Sequence Surveyor</h4><p>This tool was developed at the Computer Graphics departmartment at the University of Wisconsin-Madison as a scalable solution for comparing and clustering alignments. It uses gradients and cluster averaging to provide a way to compare multiple sequence alignments like in the <em>Lactobacillus</em> example, without the spiderweb of edges required to show congruence/inversion. This tool actually actually has interesting applications in text corpus analysis, but that is a discussion for a separate post.</p>
<p><img src="/blog/images/sequencesurv.gif" alt="SeqSurv"></p>
<h4 id="BactoGENIE"><a href="#BactoGENIE" class="headerlink" title="BactoGENIE"></a>BactoGENIE</h4><p>This project is interesting because it unifies a hardware visualization solution with a custom visual grammar. BactoGENIE was created for visualizing metagenomic datasets, which are sequencing data taken from a variety of species populating a certain local biome (water in a lake, your intestines, etc.) Figuring out similarities between all the different types of bacterial DNA in the sample yields powerful insights about the stability and the history a local biomic environment. BactoGENIE runs on a large immersive display at the Experimental Visualization Laboratory at the University of Chicago </p>
<p><img src="/blog/images/bactogenie.jpg" alt="Bact"></p>
<h3 id="Heatmap-Visualization"><a href="#Heatmap-Visualization" class="headerlink" title="Heatmap Visualization"></a>Heatmap Visualization</h3><p>Heatmap visualizations are utilized in genomics in a couple of different ways. </p>
<h4 id="IGV"><a href="#IGV" class="headerlink" title="IGV"></a>IGV</h4><h4 id="JuiceBox"><a href="#JuiceBox" class="headerlink" title="JuiceBox"></a>JuiceBox</h4><h3 id="Circular-Visualization"><a href="#Circular-Visualization" class="headerlink" title="Circular Visualization"></a>Circular Visualization</h3><h4 id="Circos"><a href="#Circos" class="headerlink" title="Circos"></a>Circos</h4><h4 id="Anvi’o"><a href="#Anvi’o" class="headerlink" title="Anvi’o"></a>Anvi’o</h4><h3 id="Network-Visualization"><a href="#Network-Visualization" class="headerlink" title="Network Visualization"></a>Network Visualization</h3><h4 id="Cytoscape"><a href="#Cytoscape" class="headerlink" title="Cytoscape"></a>Cytoscape</h4><h4 id="Ayasdi"><a href="#Ayasdi" class="headerlink" title="Ayasdi"></a>Ayasdi</h4><h4 id="Valence"><a href="#Valence" class="headerlink" title="Valence"></a>Valence</h4><h3 id="Other"><a href="#Other" class="headerlink" title="Other"></a>Other</h3><h4 id="DNA-Walks"><a href="#DNA-Walks" class="headerlink" title="DNA Walks"></a>DNA Walks</h4><h4 id="Phinch"><a href="#Phinch" class="headerlink" title="Phinch"></a>Phinch</h4>
      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>

  



    </main>

    <footer class="site-footer">
  <p class="site-info">
    Proudly powered by <a href="https://hexo.io/" target="_blank">Hexo</a> and
    Theme by <a href="https://github.com/CodeDaraW/Hacker" target="_blank">Hacker</a>
    </br>
    
    &copy; 2016 John Doe
    
  </p>
</footer>
    
  </div>
</div>
</body>
</html>