<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Structure and Function Pt 3: Building a Data Ontology | Aprameya&#39;s Thesis Blog</title>

  
  <meta name="author" content="John Doe">
  

  
  <meta name="description" content="Once we did a bit of work establishing the blind spots of current data repositories hosting structure and function data, we worked to crystallize the ">
  

  
  
  <meta name="keywords" content="">
  

  <meta id="viewport" name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no, minimal-ui">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">

  <meta property="og:title" content="Structure and Function Pt 3: Building a Data Ontology"/>

  <meta property="og:site_name" content="Aprameya&#39;s Thesis Blog"/>

  
  <meta property="og:image" content="/blog/favicon.ico"/>
  

  <link href="/blog/favicon.ico" rel="icon">
  <link rel="alternate" href="/blog/atom.xml" title="Aprameya&#39;s Thesis Blog" type="application/atom+xml">
  <link rel="stylesheet" href="/blog/css/style.css" media="screen" type="text/css">
</head>


<body>
<div class="blog">
  <div class="content">

    <header>
  <div class="site-branding">
    <h1 class="site-title">
      <a href="/blog/">Aprameya&#39;s Thesis Blog</a>
    </h1>
    <p class="site-description"></p>
  </div>
  <nav class="site-navigation">
    <ul>
      
        <li><a href="/">主页</a></li>
      
        <li><a href="/archives">归档</a></li>
      
    </ul>
  </nav>
</header>

    <main class="site-main posts-loop">
    <article>

  
    
    <h3 class="article-title"><span>Structure and Function Pt 3: Building a Data Ontology</span></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/" rel="bookmark">
        <time class="entry-date published" datetime="2017-02-08T18:31:26.000Z">
          2017-02-08
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>Once we did a bit of work establishing the blind spots of current data repositories hosting structure and function data, we worked to crystallize the principles that would inform our newly developed organizational paradigm for storing, browsing, and visualizing collections of structure and function data.<br>These principles were inspired both by the desire to go beyond currently available organzitional schema, as well as the desire to constrain our problem into a manageable enough scope to make the development of a prototype interface viable.</p>
<h2 id="Visual-Brainstorming"><a href="#Visual-Brainstorming" class="headerlink" title="Visual Brainstorming"></a>Visual Brainstorming</h2><p>As I did research about CHiP-seq, conformation capture, and the various range of cell types that might populate our prototypical library, I kept a journal where I kept note of potentially useful organizational heuristics and sketched out my mental model of data ontology. Gradually pulling a visual schema out of this organizational thinking exercise connects to some of the higher level goals of this project for me, namely using visualization to traverse high dimensional data-spaces. Here are some of my initial notes:</p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/sketch_1.JPG" alt="sketch_1" title="sketch_1">
<p>And here are more notes. I specifically tinkered with how to create a visual model describing how structure data-space and function data space might interact with one another, and how the varying layers of complexity on each side might affect those interactions. Using this nested two-panel format to describe the data space and the challenges of navigating it ended up being a great tool for communicating with Noah about the mental models used to formulate structure-function hypotheses.</p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/sketch_2_edit.png" alt="sketch_2" title="sketch_2">
<p>One crucial developmental point was coming to the conclusion that we needed ways to organize experiments both by what we know about them (e.g. attributes like phenotype, method, quality, date) as well as by direct computational comparison of experimental data which might capture similarity we <strong>don’t</strong> know about. It turns out many CHiP-seq experiments are done to capture binding affinities of proteins which have completely unknown function. If there is a protein with known function, and it has a similar binding pattern as a protein of unknown function, this a meaningful relationship we’d like to know about! Is there a way to make obvious such connections through a visual paradigm?</p>
<h3 id="Organizing-Curating-Structure-Data"><a href="#Organizing-Curating-Structure-Data" class="headerlink" title="Organizing + Curating Structure Data"></a>Organizing + Curating Structure Data</h3><p>The refined version of our left hand “structure” panel in the data ontology looks like this: </p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/struct_ontology.png" alt="struct_ontology" title="struct_ontology">
<p>Next, I tried to pin down actual quantitative measures of complexity that we would be dealing with when creating an interface to accomodate this ontology. One key constraint we decided on at this stage was to narrow our cell lines of interest to the 7 lines outlined in <a href="http://www.cell.com/fulltext/S0092-8674(14)01497-4" target="_blank" rel="external">Rao et. al 2014</a>, which is one of the most cited conformation capture experiments. The 7 cell lines for which Rao et. al 2014 has structural data are lines that have a plethora of functional experiments available (namely CHiP-seq), and furthermore the actual conformation capture done in this publication is regarded to be of very high quality. These data will be the inputs for researchers like Noah who experiment with various computational geometry techniques to generate polymer models of DNA, which constitute the actual structure files. There’s actually no scientific consensus yet on how to create these polymer models, so our hierarchy should allow for multiple structure simulations to be linked to a “parent” conformation capture dataset. Our structural hierarchy should look something like this quantitatively:</p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/struct_numbers.png" alt="struct_numbers" title="struct_numbers">
<h3 id="Organizing-Curating-Function-Data"><a href="#Organizing-Curating-Function-Data" class="headerlink" title="Organizing + Curating Function Data"></a>Organizing + Curating Function Data</h3><p>THe refined version of our right hand “function” panel in the data ontology looks like this:</p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/func_ontology.png" alt="func_ontology" title="func_ontology">
<p>Because we are interested in structure-function matches, we applied the same 7 cell line constraint from the Rao et. al 2014 paper that we used on the function side. In terms of experiment type, we are initially only going to look at organizational methods for CHiP-seq, since that seems to be the type with the largest amount of variety for our 7 cell lines. The most fine grained data type on the function side is the peak file, which is basically a histogram describing a variable quantity across the length of a genome. On ENCODE, it turns out just about every CHiP-seq experiment has multiple peak files, most of which are ‘replicates’ which ensure the reads are repeatable for a given experiment. While there are some time-dimensional experiments for which multiple peak files are an interesting level of data variety, we are going to constrain ourselves to 1 peak file per experiment to manage our complexity. Same as the CHiP-seq experiment type constraint, once we build the architecture to organize this, we might be able to expand it to accomodate more. Here’s what our hierarchy looks like:</p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/func_numbers.png" alt="func_numbers" title="func_numbers">
<h2 id="Unifying-Data-Spaces"><a href="#Unifying-Data-Spaces" class="headerlink" title="Unifying Data Spaces"></a>Unifying Data Spaces</h2><p>Bringing these two panels together as seen in the earlier notes helps us to illustrate the more nuanced conceptual problems of traversing current data repositories to observe interesting structure function relationships. </p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/struct_func.png" alt="struct_func" title="struct_func">
<p>Noah referred earlier to an ideal interface being able to expose biologically meaningful matches between structure and function. Our visual model gives an idea of what types of complexity need to be wrangled in order to allow these matches. To continue our brainstorming we situated some of the challenges and complexities on the two panel model itself.</p>
<img src="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/struct_func_issues.png" alt="struct_func_issues" title="struct_func_issues">
<p>We notice there are lot more issues implicating organization of structural data, which makes sense since structural genomics is comparatively much newer than other methods of functional genomics. </p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>

  



    </main>

    <footer class="site-footer">
  <p class="site-info">
    Proudly powered by <a href="https://hexo.io/" target="_blank">Hexo</a> and
    Theme by <a href="https://github.com/CodeDaraW/Hacker" target="_blank">Hacker</a>
    </br>
    
    &copy; 2017 John Doe
    
  </p>
</footer>
    
  </div>
</div>
</body>
</html>