<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Aprameya&#39;s Thesis Blog</title>

  
  <meta name="author" content="John Doe">
  

  

  

  <meta id="viewport" name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no, minimal-ui">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">

  

  <meta property="og:site_name" content="Aprameya&#39;s Thesis Blog"/>

  
  <meta property="og:image" content="/blog/favicon.ico"/>
  

  <link href="/blog/favicon.ico" rel="icon">
  <link rel="alternate" href="/blog/atom.xml" title="Aprameya&#39;s Thesis Blog" type="application/atom+xml">
  <link rel="stylesheet" href="/blog/css/style.css" media="screen" type="text/css">
</head>


<body>
<div class="blog">
  <div class="content">

    <header>
  <div class="site-branding">
    <h1 class="site-title">
      <a href="/blog/">Aprameya&#39;s Thesis Blog</a>
    </h1>
    <p class="site-description"></p>
  </div>
  <nav class="site-navigation">
    <ul>
      
        <li><a href="/">Home</a></li>
      
        <li><a href="/archives">Archives</a></li>
      
    </ul>
  </nav>
</header>

    <main class="site-main posts-loop">
    
  <article>

  
    
    <h3 class="article-title"><a href="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/"><span>Structure and Function Pt 3: Building a Data Ontology</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2017/02/08/Structure-and-Function-Pt-3-Building-a-Data-Ontology/" rel="bookmark">
        <time class="entry-date published" datetime="2017-02-08T18:31:26.000Z">
          2017-02-08
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        
      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/blog/2016/10/30/Structure-and-Function-Pt-2-Data-of-Interest/"><span>Structure and Function Pt. 2: Data of Interest</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2016/10/30/Structure-and-Function-Pt-2-Data-of-Interest/" rel="bookmark">
        <time class="entry-date published" datetime="2016-10-30T17:41:53.000Z">
          2016-10-30
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>After identifying the nexus of structural genomic data and functional genomic data as the problem space for which a visualization would ideally provide organization of knowledge, we had to pick some collection of datasets to impose our organizational regime. One of the first internet resources Noah pointed me to when explaining the problem of matching datasets was Cistrome dataset browser, which is a website that aggregates many types of next generation sequencing experiments on a variety of biological sources. Here’s a screenshot of Cistrome:</p>
<img src="/blog/2016/10/30/Structure-and-Function-Pt-2-Data-of-Interest/cistrome.png" alt="cistrome" title="cistrome">
<p>Noah showed me this dataset browser to give me an example of an organizational paradigm that was designed for targeted specific querying rather than exploratory querying. In the Cistrome broswer, the user has two options to narrow down the dataset library to find interesting groups of structural and functional data: select one of several hundred arcane propietary cell-line serial numbers, or from a list of ‘factors’ which identify binding sites of ChIP-seq experiments. While the cell-line serial and the ChIP binding factor are definitely identifying characteristics of an experiment, these characteristics are of little utility to a user who is perusing the library looking for interesting connections between datasets, without prior knowledge of how a given ChIP target or cell-line might be implicated in the context of a larger scientific hypothesis. </p>
<p>In the last post, I wrote about how Noah mentioned this library would be substantially more useful if it was organized in a biologically meaningful way, which provided heuristics for a user to narrow down these cell types and factors of interest. Evaluating Noah’s criticism of this particular interface drew my attention to the taxonomic aspect of this visualization challenge: how could visualization be leveraged to organize attributes like “source” and “factor” into useful mental models? Furthermore, how could these mental models allow for a navigation of structure/function data libraries in a way that facilitates hypothesis generation?</p>
<p>The next interface we explored was ENCODE. The level of organization is substantially superior to that of Cistrome. There are multiple filter types and organizational parameters that a user can leverage to find groups of experiments that are relevant to hypotheses involving heterogeneous genomic data types. Here is a screenshot of the ENCODE ‘experiment matrix’, which provides an overview of all the filtering parameters available to a user who might not know initially which specific experiment they are interested in:</p>
<img src="/blog/2016/10/30/Structure-and-Function-Pt-2-Data-of-Interest/encodematrix.png" alt="encode" title="encode">
<p>This level of organization is great, but its emphasis on hosting data types for all types of genomics workflows makes it tricky to use for structure-function specific explorations of the data space. For starters, conformation experiments have only been done on around a dozen cell types–this experiment matrix shows so many cell types for which no conformation exists at all. Additionally, since there are so many different experiment types, there aren’t any filtering parameters that allow the user to sort through CHiP-seq experiments *. ENCODE’s lack of granularity in the CHiP binding target field unfortunately makes the 1000’s of CHiP experimments available for relevant cell lines as difficult to browse as the CHiP experiments in Cistrome.</p>
<p>Working with Noah to identify why these currenly existing resources are cumbersome clarified for me the type of data ontology an ideal experiment library interface would support. Just by looking at these two website we were able to identify some crucial desired characteristics:</p>
<ul>
<li>biologically meaningful organization paradigm</li>
<li>support for organizing CHiP experiments</li>
<li>curated data showing experiments only for cell types that have an existing conformation capture</li>
<li>support for exploratory querying rather than specific querying</li>
</ul>
<p>From this list of characterstics, I worked to develop a mental model that would inform our new organizational paradigm. The next post will document this development.</p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/blog/2016/09/29/Criteria-of-Success/"><span>Criteria of Success</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2016/09/29/Criteria-of-Success/" rel="bookmark">
        <time class="entry-date published" datetime="2016-09-29T21:39:14.000Z">
          2016-09-29
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        
      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/blog/2016/09/28/Structure-and-Function-An-Introduction/"><span>Structure and Function: An Introduction</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2016/09/28/Structure-and-Function-An-Introduction/" rel="bookmark">
        <time class="entry-date published" datetime="2016-09-28T22:52:32.000Z">
          2016-09-28
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>As mentioned in my introductory post, my culminatory work of this thesis project will be a portfolio of 3 design solutions to unique visualization challenges in genomics. One of these design projects is a collaboration with the <a href="http://guttmanlab.caltech.edu/" target="_blank" rel="external">Guttman Laboratory for lncRNA Biology</a>, in which I will explore the creation of workflows for researchers interested in relationships between the physical structure of a genome inside a cell, and its functionality.</p>
<p>This project is an extension of some of the work I did this past summer as an intern in the JPL/Caltech Data Visualization Program. Over the summer, I worked with other interns and researchers at the Guttman Lab to understand the challenges involved in studying relationships between genomic structure and function. We learned that there are valuable insights to be uncovered through the analysis of DNA conformation data, but this analysis requires the navigation of very opaque and complicated datasets.</p>
<h2 id="Basics-of-3D-Genomics"><a href="#Basics-of-3D-Genomics" class="headerlink" title="Basics of 3D Genomics"></a>Basics of 3D Genomics</h2><p>The 3 dimensional shape of DNA inside a cell consists of entanglements that occur across several orders of magnitude. The following graphic from <a href="https://www.ncbi.nlm.nih.gov/pubmed/26226004" target="_blank" rel="external">Ea et. al 2015</a> shows the way 2 meters of DNA sequence are packaged into 5 microns of spherically confined space in the center of the nucleus. </p>
<img src="/blog/2016/09/28/Structure-and-Function-An-Introduction/conformation.png" alt="conformation" title="conformation">
<p>Understanding the unique circumstances under which a sequence will fold in different ways is critical to understanding how cells can exhibit a diverse array of behaviors driven by the same sequence, and how defects in genome structure can cause cancer and many other disease. The process of coming to such an understanding is a multistage process. Researchers must dive into conformation capture datasets, which are stored in the form of massive adjacency matrices, to observe patterns of topological association between certain segments of DNA sequence. Areas of interest must then be cross referenced against expermental data which describes functional behavior of regions of DNA sequence. Other researchers approach this problem in the other direction: a particular bit of sequence is interesting for its functional characterstics, so they want to see how that region of sequence conforms in 3d space under certain conditions. The unification of structural and functional data allows the researcher to formulate models for <strong>the general principles that determine 3D genome structure</strong> and <strong>relationships between 3d genome structure and specific cellular function</strong>. </p>
<h2 id="Creating-a-Workflow"><a href="#Creating-a-Workflow" class="headerlink" title="Creating a Workflow"></a>Creating a Workflow</h2><p>Bringing these two types of datasets together in a way that facilitates analysis and developing a <em>coherent workflow</em> for meaningful hypothesis generation is currently very difficult. The main hurdles to such a workflow are:</p>
<ul>
<li>The lack of tools to visualize and interact with 3D Genome structures. This was the subject of my summer work with the Guttman Lab which yielded <a href="http://gin.caltech.edu/" target="_blank" rel="external">Gin</a></li>
<li>The lack of any sort of organizational paradigm for the massive amounts of functional genomic data that can be mapped to structural information. This will be the challenge I intend to take on as a dedicated portion of my thesis project.</li>
</ul>
<p>My collaborator at the Guttman Lab, Noah Ollikainen, elaborates further on this second point:</p>
<blockquote>
<p>Current websites for accessing functional genomic data allow users to view gene expression data (RNA-seq) as well as find the DNA binding sites for proteins that regulate gene expression (ChIP-seq). A major challenge with using these websites is navigating the sheer volume of data, as RNA-seq experiments have now been performed in hundreds of different cell types and ChIP-seq experiments been performed for hundreds of different proteins. To access these data, a user must know exactly what they are looking for, e.g., the DNA binding sites of protein A in the genome of human cell-type B, and this severely limits our ability to <strong>perform exploratory research that enables novel discoveries</strong>. </p>
<p>An improved interface would organize these data in a biologically meaningful way. For example, these data could be organized hierarchically, where experiments in cell types derived from the same organ (e.g., liver, brain, kidney) are grouped together and experiments on proteins with a shared function (gene activator, gene repressor) are grouped together. Additionally, filters could be implemented to allow the user to restrict the data to a subset of interest, for example, cell lines derived from females only or data collected within the past N years. This organizational scheme would allow the user to easily select the most relevant functional genomic data for their research and explore the general relationship between these data and 3D genome structure.</p>
</blockquote>
<p>Noah’s emphasis on exploratory research harkens back to the criteria of success I tackled in a previous blog post. One of the goals of genomic data visualization is to allow the exploration of complex data spaces in meaningful ways. In many cases, researchers don’t know what they are looking for, and look to visualization as a means for developing heuristics for narrowing down and identifying signficant features in these complex data spaces. The ideal workflow would combine methods of visualizing structure and methods of identifying functional data of interest to create a holistic understanding. I’ve made the following diagram to sketch out our goals here:</p>
<img src="/blog/2016/09/28/Structure-and-Function-An-Introduction/FunctionCartoon.png" alt="function" title="function">

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/blog/tags/StructureProject/">StructureProject</a>
    </span>
    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/blog/2016/09/18/background/"><span>Background &amp; Prior Work</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2016/09/18/background/" rel="bookmark">
        <time class="entry-date published" datetime="2016-09-18T23:27:58.000Z">
          2016-09-18
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>There are several different types of media that need to be explored to consolidate a history of prior work in genomics visualization. </p>
<ul>
<li>media consisting of visualizations that describe a problem, process, or observation relating to genomic data.</li>
<li>media describing the problem space of visualization in genomics. this has to do with genomics’ place within the epistemological regime of systems biology </li>
<li>media that attempts to create criteria for good/bad visualizations</li>
</ul>
<p>In this post I will tackle the first area. I’ve tried to select visualizations that I think are either effective or ineffective for notable reasons; there are a lot of bad visualizations out there so it is important to learn from ‘near misses’ instead of trying to extract meaning from something that is a complete failure. </p>
<p><a href="https://petridishtalk.com/2012/06/30/gene-visualization/" target="_blank" rel="external">This</a> blog post by Mo Rahman, the founder of <a href="http://genedrop.com/" target="_blank" rel="external">GeneDrop</a>, provides a good overview of the most common types of visual grammar used in contemporary genomics problem solving. I want to elaborate on this overview of genomics visualization strategies by organizing visualizations by their functional goals in addition to just their overt appearance. </p>
<p>Furthermore, I will be covering only non-generic visualizations </p>
<h3 id="Track-Visualization"><a href="#Track-Visualization" class="headerlink" title="Track Visualization"></a>Track Visualization</h3><p>Track visualizations present sequential information in a horizontal extended track, and are primarily utilized to examine issues of alignment between multiple sequences. The genome alignment problem space ranges from analyzing small varations in sequence amongst members of a population or species, to asking questions about complex conserved trends across species or generations. Given the scale and complexity of the variety of alignment relationships potentially of interest, it can be very trickly to legibly visualize an observation or a hypothesis. The following figure shows similarity and inversion of particular regions across 11 <em>Lactobacillus</em> genomes represented as tracks, from the paper <a href="http://journal.frontiersin.org/article/10.3389/fmicb.2012.00048/full" target="_blank" rel="external">“Tracing lifestyle adaptation in prokaryotic genomes”</a>. It gets quite messy.</p>
<p><img src="/blog/images/alignmentmess.jpg" alt="alignmentmess"></p>
<p>As such, researchers have explored custom track visualization solutions when a linear genome representation is required. A couple interesting examples include:</p>
<h4 id="GeneDrop-Custom-Browser"><a href="#GeneDrop-Custom-Browser" class="headerlink" title="GeneDrop Custom Browser"></a>GeneDrop Custom Browser</h4><p>This browser has been shown by Mo Rahman on Twitter and on his blog. It’s a custom in house genome browser used for informatics work at GeneDrop, and effectively shows how a well designed track visualization can frame the complexity at various levels of detail present in a multiple sequence alignment analysis. Here it is shown in a workflow for resolving structural variations in certain repetetive regions of genome sequence</p>
<p><img src="/blog/images/GeneDropProprietaryBrowser.png" alt="GeneDrop"></p>
<p>and again here in an interesting <a href="https://petridishtalk.com/2014/10/02/retooling-analysis-pipelines-from-human-to-ebov-ngs-data-for-rapid-alignment-and-strain-identification/" target="_blank" rel="external">post</a> about analyzing strain variations of the ebola virus</p>
<p><img src="/blog/images/GeneDrop2.png" alt="GeneDrop2"></p>
<h4 id="Sequence-Surveyor"><a href="#Sequence-Surveyor" class="headerlink" title="Sequence Surveyor"></a>Sequence Surveyor</h4><p>This tool was developed at the Computer Graphics department at the University of Wisconsin-Madison as a scalable solution for comparing and clustering alignments. It uses gradients and cluster averaging to provide a way to compare multiple sequence alignments like in the <em>Lactobacillus</em> example, without the spiderweb of edges required to show congruence/inversion. This tool actually actually has interesting applications in text corpus analysis, but that is a discussion for a separate post.</p>
<p><img src="/blog/images/sequencesurv.gif" alt="SeqSurv"></p>
<h4 id="BactoGENIE"><a href="#BactoGENIE" class="headerlink" title="BactoGENIE"></a>BactoGENIE</h4><p>This project is interesting because it unifies a hardware visualization solution with a custom visual grammar. BactoGENIE was created for visualizing metagenomic datasets, which are sequencing data taken from a variety of species populating a certain local biome (water in a lake, your intestines, etc.) Figuring out similarities between all the different types of bacterial DNA in the sample yields powerful insights about the stability and the history a local biomic environment. BactoGENIE runs on a large immersive display at the Experimental Visualization Laboratory at the University of Chicago </p>
<p><img src="/blog/images/bactogenie.jpg" alt="Bact"></p>
<h4 id="UCSC"><a href="#UCSC" class="headerlink" title="UCSC"></a>UCSC</h4><p>The <a href="genome.ucsc.edu">UCSC genome browser</a> provides the most widely used and most accessible track visualization platform for genomic data. This project began to showcase the initial findings of the Human Genome Project, and continues to be a widely used repository for genomic alignment and annotation data for several different species. The UCSC browser actually combines data that are traditionally seen in heatmaps (such as measurement of gene expression, protein binding affinity, etc) along the distance of the sequence in addition to the tracks which indicate locations of genes/promoter regions for various species. While seeing these two types of information alongside each other can be very powerful analytically, the sheer quantity of informational tracks available on UCSC makes it difficult to explore potentially interesting patterns across a sequence. As such, the UCSC browser is used primarily to look up sections of sequence that have already been identified as noteworthy. </p>
<p><img src="/blog/images/ucsc.JPG" alt="alt"></p>
<h3 id="Heatmap-Visualization"><a href="#Heatmap-Visualization" class="headerlink" title="Heatmap Visualization"></a>Heatmap Visualization</h3><p>Heatmap visualizations are used for a variety of purposes in genomic analysis. When understanding the range of gene products created by a certain genome, researchers will use heatmaps to visualize patterns occuring in various forms of microarray data across specimens. This heatmap from <a href="http://journal.frontiersin.org/article/10.3389/fonc.2012.00065/full" target="_blank" rel="external">Allard et. al 2012</a> shows clustered patterns of 66 transcripts differentially expressed between two populations in a cancer study.</p>
<p><img src="/blog/images/expressionheatmap.jpg" alt="Express"></p>
<h4 id="IGV"><a href="#IGV" class="headerlink" title="IGV"></a>IGV</h4><p>The <a href="http://software.broadinstitute.org/software/igv/" target="_blank" rel="external">Integrative Genomics Viewer</a> is a very widely used interface for aggregating and organizing heatmap data. It was created by the Broad Institute in 2011 to allow analysis of diverse datasets that annotate a genomic sequence across multiple scales. </p>
<p><img src="/blog/images/igv_cherry.gif" alt="igv"> </p>
<p>Heatmaps are also used to show relationships of adjacency and connectedness. Additional explorations into microarray data like the one above look not only at how genes are differentially expressed amongst various specimens in an experiment, but how genes are coexpressed with other genes. In this way the heatmap is a reprojection of a graph whereby the rows and columns are the nodes and the color value in the cells represents strength of links. This graphic from <a href="http://circgenetics.ahajournals.org/content/4/1/26.full" target="_blank" rel="external">Dewey et. al 2010</a> shows such a reprojection.</p>
<p><img src="/blog/images/coexpression.jpg" alt="coexp"></p>
<p>Another way heatmaps are used to project data about adjacency is in studies of chromosomal conformation. Recent technological breakthroughs have allowed researchers to capture the 3 dimensional structure of a genome inside a cell with higher degrees of confidence, and the visualization of this data is driving the study of how genomic structure affects its function. This graphic from <a href="http://www.cell.com/fulltext/S0092-8674(14)01497-4" target="_blank" rel="external">Rao et. al 2014</a> utilizes heatmaps in such a way to explain folding of chromatin loops.</p>
<p><img src="/blog/images/hic.jpg" alt="hic"></p>
<h4 id="JuiceBox"><a href="#JuiceBox" class="headerlink" title="JuiceBox"></a>JuiceBox</h4><p><a href="http://www.aidenlab.org/juicebox/" target="_blank" rel="external">Juicebox</a> is a software for visualizing datasets that describe structural conformation of genomes. Like IGV, it provides dynamic rescaling of data at different zoom levels, which allows a user to observe interesting features across various orders of magnitude. It is currently one of the premier toolkits for structure-function research in genomics.</p>
<p><img src="/blog/images/straight_edge.png" alt="juicebox"></p>
<h3 id="Tree-Visualization"><a href="#Tree-Visualization" class="headerlink" title="Tree Visualization"></a>Tree Visualization</h3><p>Visualizations describing the structure of biological classifications or ‘trees of life’ are amongst the most nascent forms of scientific visualization. Phylogenetic and taxonomic visualizations showcase evolutionary relationships amongst a population of biological entities. The visualized phylogeny provides useful information for a variety of populations, which could include microbial communities local to a biome, families of proteins present across species, or even more broadly defined groups in which relationships of common ancestry and heredity are of interest. Modern approaches to visualizing phylogenies typically utilize circular dendrograms, which provide increased space for more granular classification at the edges. </p>
<h4 id="GraPhlAn"><a href="#GraPhlAn" class="headerlink" title="GraPhlAn"></a>GraPhlAn</h4><p><a href="https://huttenhower.sph.harvard.edu/graphlan" target="_blank" rel="external">GraPhlAn</a> is a genome phylogeny and taxonomy visualization package based around circular dendrograms. This particular figure from the GraPhlAn paper, <a href="https://peerj.com/articles/1029/" target="_blank" rel="external">Asnicar et. al 2015</a>, demonstrates how the radial structure of the tree can be used to encode both information about hierarchy in a population of bacterial genomes, as well as functional attributes characterizing different regions of the genomic hierarchy.</p>
<p><img src="/blog/images/phylotree.jpg" alt="phylo"></p>
<p>Earlier in the paper the authors present this meta-diagram which demonstrates the process by which a dendrogram can be generated from a hierarchical data source. These types of meta-diagrams give an interesting peek into the construction of a design vocabulary for a given visualization problem, and will be covered more extensively in a future blog post.</p>
<p><img src="/blog/images/phylometa.png" alt="phylometa"></p>
<h4 id="Anvi’o"><a href="#Anvi’o" class="headerlink" title="Anvi’o"></a>Anvi’o</h4><p><a href="http://merenlab.org/software/anvio/" target="_blank" rel="external">Anvi’o</a> is a circular dendogram visualization designed as part of a metagenomics analysis pipeline. This pipeline is used to explore how genomic changes across multiple species in an environmental sample are related to broader environmental changes. The visualizations generated by Anvi’o are similar to the dendrograms seen in GraPhlAn, with a notable difference being Anvi’o is designed for encoding substantially more functional information by using the radial axis to represent time.</p>
<p><img src="/blog/images/anvio.png" alt="anvio"></p>
<h3 id="Network-Visualization"><a href="#Network-Visualization" class="headerlink" title="Network Visualization"></a>Network Visualization</h3><p>There are a multitude of uses for network visualizations in genomics. Connectivity is a very important theme across genomics, and questions of connectedness are frequent avenues for interrogating a variety of different genomic data. As such, projection and reprojection of various types of genomic data into node-link space is common practice in contemporary genomic visual grammar. Furthermore, networks are intimately related to the epistemological task of genomics within the regime of systems biology: how do characterstics of the genome give rise to function and behavior of other biological systems?   </p>
<h4 id="Circos"><a href="#Circos" class="headerlink" title="Circos"></a>Circos</h4><p><a href="http://www.circos.ca/" target="_blank" rel="external">Circos</a> is a visualization package that is used to make a multitude of chord diagram based plots. Although it is purportedly agnostic to data source, it was designed as a utility for visualizing connectivity and adjacency in genomic data. I see the chord diagram as a reprojected heatmap. The following figure from <a href="http://link.springer.com/article/10.1007/s11032-013-9941-x" target="_blank" rel="external">Wei et. al</a> shows a circos plot to describe connectivity amongst of group of protein kinases in maize. There are multiple types of functional annotation for each protein kinase visualized radially on the chord diagram as well.</p>
<p><img src="/blog/images/circosmaize.png" alt="circosmaize"></p>
<h4 id="Ayasdi"><a href="#Ayasdi" class="headerlink" title="Ayasdi"></a>Ayasdi</h4><p>Networks visualizations can also be utilized as a low dimensional embedding of a clustering operation performed in high dimensional space. <a href="https://www.ayasdi.com/blog/office-of-the-cdo/when-is-breast-cancer-like-ovarian-cancer/" target="_blank" rel="external">Ayasdi</a> utilizes network visualizations to describe the results of topological data analysis performed on high dimensional data of various types. The figure here is comparing clustering of genomic expression profiles of various tumor biopsy samples. The relative shape and color of each network gives a viewer an idea of how similar any two tumors might be for a variety of different cancers. Between the node-link structure and the color gradient, a lot of information can be embedded in this visual style.</p>
<p><img src="/blog/images/breast_cancer_data1.png" alt="ayasdi"></p>
<h4 id="Valence"><a href="#Valence" class="headerlink" title="Valence"></a>Valence</h4><p>Valence was a design experiment that was part of Ben Fry’s Master’s Thesis: Organic Information Design. Valence sought to explore how networks can function as reprojections of linear sequences, and can potentially elucidate questions about connectedness of phrases that are not apparent in more traditional track based visualizations. It was utilized both on large corpuses of text as well as genomic data.</p>
<p><img src="/blog/images/valence.jpg" alt="valence"></p>
<h4 id="KEGG"><a href="#KEGG" class="headerlink" title="KEGG"></a>KEGG</h4><p>From the Kyoto Encyclopedia of Genes and Genomes website, a KEGG pathway is desribed as a manually drawn map representing knowledge of molecular interaction and reaction networks linking data on genomics, transcriptomics, proteomics, and metabolomics. These pathway maps describe higher order systemic functions which implicate genomics into the larger descriptive paradigms of systems biology. While they do not provide the same type of exact mapping from data space to visual space that other modes of genomic visualization do, they are still very important as heuristic tools in systems biology. The following diagram is of KEGG pathway hsa01100, which describes the sum total of metabolic processes that drive human cellular activity.  </p>
<p><img src="/blog/images/kegg_global.png" alt="kegg"></p>
<h3 id="Multi-Visualization-Suites"><a href="#Multi-Visualization-Suites" class="headerlink" title="Multi Visualization Suites"></a>Multi Visualization Suites</h3><p>Sometimes being able to switch between multiple views of a dataset in succession is useful for understanding distributional characteristics and ascertaining significance of outliers. </p>
<h4 id="Phinch"><a href="#Phinch" class="headerlink" title="Phinch"></a>Phinch</h4><p><a href="http://phinch.org/" target="_blank" rel="external">Phinch</a> was created by the data visualization studio Pitch Interactive as a tool for visualizing high throughput metagenomic datasets, which contain genomic data from a variety of different organisms present in a local environment. In addition to providing a curated selection of plots to dive into a given dataset, particular attention is paid to providing a ‘designed’ data load in menu. Using data visualization to allow a user to identify which components of a vast data repository are to be visualized in further detail is a interesting and widely useful application of visual grammar in genomics. </p>
<p><img src="/blog/images/viz_gallery_phinch.png" alt="phinch"></p>
<h4 id="Isometric-Haplotype-Blocks"><a href="#Isometric-Haplotype-Blocks" class="headerlink" title="Isometric Haplotype Blocks"></a>Isometric Haplotype Blocks</h4><p>Ben Fry designed this utility as part of his PhD thesis on computational information design. The study of haplotypes, which are groups of polymorphisms in the genome of a species that are often found together, is critical to understanding what types of genetic variation are related to disease. Understanding haplotype membership can be tricky, and requires identifying correlations amongst groups of polymorphisms in a population that pass a certain threshold known as Linkage Disequilibrium (LD). If there is no significant mechanism causing associations between polymorphisms in one region of a genome and another, population data should reflect that all combinations of polymorphisms in region A and region B are equally probable to occur in a specimen through normal genetic variation. However if region A and B are in strong LD, specific types of polymorphisms in region A are strongly correlated with specific polymorphisms in region B.</p>
<p>Fry’s visualization is an exercise in understanding how data on polymorphism and LD in a given population can help construct an idea of haplotype clustering for that population. By providing multiple isomorphic views of the dataset, the user can maintain context while considering different types of comparative evaluations of the patterns of polymorphism.</p>
<p><img src="/blog/images/isometricblocks-900.jpg" alt="isometric"></p>
<h3 id="Other"><a href="#Other" class="headerlink" title="Other"></a>Other</h3><h4 id="DNA-Walks"><a href="#DNA-Walks" class="headerlink" title="DNA Walks"></a>DNA Walks</h4>
      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    

    </div>

    
  </div>
</article>



  <article>

  
    
    <h3 class="article-title"><a href="/blog/2016/09/17/hello-world/"><span>Hello World</span></a></h3>
    
  

  <div class="article-top-meta">
    <span class="posted-on">
      <a href="/blog/2016/09/17/hello-world/" rel="bookmark">
        <time class="entry-date published" datetime="2016-09-17T20:57:41.000Z">
          2016-09-17
        </time>
      </a>
    </span>
  </div>


  

  <div class="article-content">
    <div class="entry">
      
        <p>Welcome to <a href="https://hexo.io/" target="_blank" rel="external">Hexo</a>! This is your very first post. Check <a href="https://hexo.io/docs/" target="_blank" rel="external">documentation</a> for more info. If you get any problems when using Hexo, you can find the answer in <a href="https://hexo.io/docs/troubleshooting.html" target="_blank" rel="external">troubleshooting</a> or you can ask me on <a href="https://github.com/hexojs/hexo/issues" target="_blank" rel="external">GitHub</a>.</p>
<h2 id="Quick-Start"><a href="#Quick-Start" class="headerlink" title="Quick Start"></a>Quick Start</h2><h3 id="Create-a-new-post"><a href="#Create-a-new-post" class="headerlink" title="Create a new post"></a>Create a new post</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ hexo new <span class="string">"My New Post"</span></div></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/writing.html" target="_blank" rel="external">Writing</a></p>
<h3 id="Run-server"><a href="#Run-server" class="headerlink" title="Run server"></a>Run server</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ hexo server</div></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/server.html" target="_blank" rel="external">Server</a></p>
<h3 id="Generate-static-files"><a href="#Generate-static-files" class="headerlink" title="Generate static files"></a>Generate static files</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ hexo generate</div></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/generating.html" target="_blank" rel="external">Generating</a></p>
<h3 id="Deploy-to-remote-sites"><a href="#Deploy-to-remote-sites" class="headerlink" title="Deploy to remote sites"></a>Deploy to remote sites</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ hexo deploy</div></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/deployment.html" target="_blank" rel="external">Deployment</a></p>

      
    </div>

  </div>

  <div class="article-footer">
    <div class="article-meta pull-left">

    

    
    

    <span class="post-tags">
      <i class="icon-tags"></i>
        <a href="/blog/tags/LOL/">LOL</a>
    </span>
    

    </div>

    
  </div>
</article>




<nav class="pagination">
  
  
</nav>
    </main>

    <footer class="site-footer">
  <p class="site-info">
    Proudly powered by <a href="https://hexo.io/" target="_blank">Hexo</a> and
    Theme by <a href="https://github.com/CodeDaraW/Hacker" target="_blank">Hacker</a>
    </br>
    
    &copy; 2017 John Doe
    
  </p>
</footer>
    
  </div>
</div>
</body>
</html>